models/DataObject/Listing.php

```diff
@@ -211,4 +211,19 @@ public function addDistinct()
     {
         return false;
     }
+
+    /**
+     * @param string $field database column to use for WHERE condition
+     * @param string $operator SQL comparison operator, e.g. =, <, >= etc. You can use "?" as placeholder, e.g. "IN (?)"
+     * @param string|int|float|float|array $data comparison data, can be scalar or array (if operator is e.g. "IN (?)")
+     *
+     * @return static
+     */
+    public function filterByField($field, $operator, $data) {
```

Comment:
Maybe we should call this method `addFilterByField()` and mark it as `@internal` to be consistent in naming?